README.md
# Genesis What's New Info 
**Contributors:** daveshine, deckerweb, wpautobahn  
**Donate link:** https://www.paypal.me/deckerweb  
**Tags:** genesis, genesiswp, genesis framework, what's new, version, branch, updates, update, upgrade, admin, deckerweb  
**Requires at least:** 4.7 plus Genesis Framework  
**Tested up to:** 5.2  
**Requires PHP:** 5.6  
**Stable tag:** 1.3.4  
**License:** GPL-2.0-or-later  
**License URI:** https://opensource.org/licenses/GPL-2.0  

Easy access of the Genesis What's New Admin overview page - not only on updates but everytime. Makes sense, heh? :)

[<img src="https://ps.w.org/genesis-whats-new-info/assets/banner-1544x500.png" data-canonical-src="https://ps.w.org/genesis-whats-new-info/assets/banner-1544x500.png" width="772" height="250" />](https://wordpress.org/plugins/genesis-whats-new-info/)


## Description 

## Get Access to Genesis What's New Admin Page
This simple & lightweight plugin has currently only one task: bring you access to the "What's New" page for Genesis Framework. You can access it everytime in the left-hand admin menu under "Genesis > What's New". (Before this plugin it did only show up when you did an auto update of Genesis.)

**Please note:** The plugin requires the *Genesis Theme Framework*, a paid premium product released by StudioPress/ WPEngine, Inc. (via studiopress.com).


### Video of Plugin's Live Demo and Walkthrough: 
[![Video of Plugin's Live Demo and Walkthrough](https://img.youtube.com/vi/w6CUGbHm5PI/0.jpg)](https://www.youtube.com/watch?v=w6CUGbHm5PI)
[**original video link**](https://www.youtube.com/watch?v=w6CUGbHm5PI) *by plugin developer David Decker*


### Requirements 
* Genesis version 2.6.x or higher (obviously :).
* At least capability of `edit_theme_options` for the user.
* ...and the user must have any of the default Genesis admin pages activated in the user profile.


### Future Roadmap 
* Maybe further info/ features will be added to this plugin from time to time.
* *Suggestions can be made via contact info below.*


### Localization 
* English (default) - always included
* German (de_DE) - always included
* German formal (de_DE_formal) - always included
* `.pot` file (`genesis-whats-new-info.pot`) for translators is also always included :)
* Easy plugin translation platform: [Translate "Genesis What's New Info"...](https://translate.wordpress.org/projects/wp-plugins/genesis-whats-new-info)


### Liked Genesis What's New Info?
* **Rate us 5 stars** on [WordPress.org](https://wordpress.org/support/plugin/genesis-whats-new-info/reviews/?filter=5/#new-post) :)
* Join our [**Facebook User Community Support Group**](https://www.facebook.com/groups/deckerweb.wordpress.plugins/)
* Like our [**Facebook Info Page for Deckerweb Plugins**](https://www.facebook.com/deckerweb.wordpress.plugins/)
* [**Subscribe to my Newsletter for insider info on this plugin**](https://eepurl.com/gbAUUn), plus tutorials and more stuff on deckerweb WordPress plugins - join a thriving community of site builders!


## Check out my other Genesis Plugins
* [**Genesis Widgetized Not Found & 404**](https://wordpress.org/plugins/genesis-widgetized-notfound/) - Easy Setup for 404 Page and Search Not Found - be prepared for the edge cases
* [**Genesis Extra Settings Transporter**](https://wordpress.org/plugins/genesis-extra-settings-transporter/) - addionally export Genesis specific plugin & child theme settings
* [**Genesis Layout Extras**](https://wordpress.org/plugins/genesis-layout-extras/) - Setup Default Layouts in Genesis for WordPress
* [Genesis Shortcode UI (for Shortcake)](https://github.com/deckerweb/genesis-shortcode-ui)
* [Genesis Elementor Canvas](https://github.com/deckerweb/genesis-elementor-canvas)
* [Genesis Featured Page Extras](https://wordpress.org/plugins/genesis-featured-page-extras/) - more options for the featured page widget
* [Genesis Prose Extras](https://wordpress.org/plugins/genesis-prose-extras/) - add-on for Prose child theme
* [Genesis Printstyle Plus](https://wordpress.org/plugins/genesis-printstyle-plus/) - just print it out - only what's needed
* [Genesis Widgetized Archive](https://wordpress.org/plugins/genesis-widgetized-archive/) - easily set up content for your Archive page (template)
* [Genesis Widgetized Footer](https://wordpress.org/plugins/genesis-widgetized-footer/) - easily set up your copyright/ credits/ back to top
* [Genesis Connect for Easy Digital Downloads](https://wordpress.org/plugins/genesis-connect-edd/) - plugin for easy & elegant integration
* [*My Genesis plugins newsletter*](https://eepurl.com/gbAUUn)


## My Other Plugins
* [**Toolbar Extras for Genesis & Elementor - WordPress Admin Bar Enhanced**](https://wordpress.org/plugins/toolbar-extras/)
* [**Toolbar Extras for Oxygen Builder - Add-On plugin**](https://wordpress.org/plugins/toolbar-extras-oxygen/)
* [**Builder Template Categories - for WordPress Page Builders**](https://wordpress.org/plugins/builder-template-categories/)
* [**Polylang Connect for Elementor – Language Switcher & Template Tweaks**](https://wordpress.org/plugins/connect-polylang-elementor/)
* [**Simple Download Manager for WP Document Revisions**](https://wordpress.org/plugins/wpdr-simple-downloads/)
* [Multisite Toolbar Additions](https://wordpress.org/plugins/multisite-toolbar-additions/)
* [Cleaner Plugin Installer](https://wordpress.org/plugins/cleaner-plugin-installer/)
* [*My plugins newsletter*](https://eepurl.com/gbAUUn)


## Installation 

**NOTE:** Only works with *Genesis Framework* as the parent theme. This is a paid premium product by StudioPress/ WPEngine, Inc., available via studiopress.com.


### Installation of Plugin: 
1. Installing alternatives:
 * *via Admin Dashboard:* Go to **Plugins > Add New**, search for "Genesis What's New Info", click "install"
 * *OR via direct ZIP upload:* Upload the ZIP package via **Plugins > Add New > Upload** in your WP Admin
 * *OR via FTP upload:* Upload `genesis-whats-new-info` folder to the `/wp-content/plugins/` directory
2. Activate the plugin through the 'Plugins' menu in WordPress
3. Go to the "Genesis" admin page and you'll see a new sub menu item "What's New" - which shows you the recent updates for your installed version branch of Genesis Framework.

**Note:** The "Genesis Framework" is required for this plugin in order to work. If you don't own a copy it yet, this premium parent theme has to be bought. More info about that you'll find here: https://ddwb.me/getgenesis


## Frequently Asked Questions 


### Where does the content come from? 
The content of the "What's New" page comes straight from Genesis core! This plugin only hooks in a few links to this *already existing* admin page.



### Why a new meta box on Genesis Settings? 
The existing "Information" meta box from Genesis core has not much useful info. The new "At a Glance" box brings more info and links, also for the active child theme.



### More info on Translations? 

* English - default, always included
* German (de_DE): Deutsch - immer dabei! :-)
* For custom and update-secure language files please upload them to `/wp-content/languages/genesis-whats-new-info/` (just create this folder) - This enables you to use fully custom translations that won't be overridden on plugin updates. Also, complete custom English wording is possible with that as well, just use a language file like `genesis-whats-new-info-en_US.mo/.po` to achieve that (for creating one see the following tools).

**Easy WordPress.org plugin translation platform with GlotPress platform:** [**Translate "Genesis What's New Info"...**](https://translate.wordpress.org/projects/wp-plugins/genesis-whats-new-info)

*Note:* All my plugins are internationalized/ translateable by default. This is very important for all users worldwide. So please contribute your language to the plugin to make it even more useful. For translating and validating I recommend the awesome ["Poedit Editor"](https://www.poedit.net/), which works fine on Windows, macOS and Linux.


## Screenshots 

### 1. Genesis What's New Info: Left-hand admin menu: new item "What's New (<branch version>)". ([Click here for larger version of screenshot](https://www.dropbox.com/s/1pvf9xhk25zgrx0/screenshot-01.png))
![Genesis What's New Info: Left-hand admin menu: new item "What's New (<branch version>)". ([Click here for larger version of screenshot](https://www.dropbox.com/s/1pvf9xhk25zgrx0/screenshot-01.png)).](https://ps.w.org/genesis-whats-new-info/assets/screenshot-01.png)


### 2. Genesis What's New Info: New link to "What's New" page in upgrade notice. ([Click here for larger version of screenshot](https://www.dropbox.com/s/r0o9oyw3vo95el8/screenshot-02.png))
![Genesis What's New Info: New link to "What's New" page in upgrade notice. ([Click here for larger version of screenshot](https://www.dropbox.com/s/r0o9oyw3vo95el8/screenshot-02.png)).](https://ps.w.org/genesis-whats-new-info/assets/screenshot-02.png)



## Changelog 

### ⚡ 1.3.4 - 2019-05-07
* *New: Successfully tested with WordPress 5.2*
* New: Integrated with WordPress 5.2+ new Site Health feature: Genesis What's New Info now has an extra section on the Debug Info tab - this is especially helpful for support requests
* New: Added Genesis Developer Docs to Metabox
* Tweak: Updated Genesis Changelog URL
* Tweak: Updated bundled library DDWlib Plugin Installer Recommendations to latest version (v1.4.1) - feature updates
* Tweak: Updated `.pot` file plus all German translations (formal, informal) and language packs
* New: [Join my newsletter for DECKERWEB WordPress Plugins](https://eepurl.com/gbAUUn) - insider info, plus tutorials and more useful stuff


### ⚡ 1.3.3 - 2018-11-03
* New: [Video of plugin walkthrough and live demo](https://www.youtube.com/watch?v=w6CUGbHm5PI)
* Tweak: Updated bundled library DDWlib Plugin Installer Recommendations to latest version (v1.2.0) - which brings enhanced CSS styles, including for the "Dark Mode" plugin
* Tweak: Few internal code tweaks and improvements


### ⚡ 1.3.2 - 2018-10-01
* New: Added plugin update message also to Plugins page (overview table)
* New: Created special [Facebook Group for user community support](https://www.facebook.com/groups/deckerweb.wordpress.plugins/) for all plugins from me (David Decker - DECKERWEB), this one here included! ;-) - [please join at facebook!](https://www.facebook.com/groups/deckerweb.wordpress.plugins/)
* Tweak: Updated bundled library DDWlib Plugin Installer Recommendations to latest version (v1.1.0) - which brings smaller additions and enhancements, like CSS styles to the upload areas and plugin cards, plus plugin version number on plugin cards
* Tweak: Small internal code improvements
* Tweak: Updated `.pot` file plus all German translations (formal, informal) and language packs


### ⚡ 1.3.1 - 2018-08-24
* Fix: Bug with wrong constant


### 🎉 1.3.0 - 2018-08-24 
* New: Added extra "At a Glance" meta box on Genesis Theme Settings with proper version info, and links to: What's New page, Changelog website, StudioPress Blog, #GenesisWP Facebook Group, #GenesisWP Slack Channel, #GenesisWP Tweets -- Note: this is fully optional, meta box can be removed via "Screen Options", and also via our filter/ WP Core function
* New: Added `composer.json` file to the plugin's root folder - this is great for developers using Composer
* New: Added `README.md` file for plugin's GitHub.com repository to make it more readable there
* Tweak: Simplied rendering of What's New Page again, hooray! :)
* Tweak: Changed all of the plugin's filter hook tags to new format!
* Tweak: Added plugins recommendations library by deckerweb to improve the plugin installer tips (old filter function got removed)
* Tweak: Updated `.pot` file plus all German translations (formal, informal) and language packs
* Tweak: Enhanced, improved and corrected Readme.txt file here - also tweaked FAQ entry



### 🎉 1.2.0 - 2018-04-20 
* New: Release on GitHub.com as well (for issues, development etc.), see here: [https://github.com/deckerweb/genesis-whats-new-info](https://github.com/deckerweb/genesis-whats-new-info)
* New: New plugin banner and icon on wordpress.org
* Fix: Make it work again, requires the newest Genesis version 2.6.0 or higher.
* Tweak: A few subtle tweaks here and there.
* Tweak: Updated translations and `.pot` file.



### ⚡ 1.1.1 - 2014-02-02 
* Fix: Fixed PHP notice.



### 🎉 1.1.0 - 2014-01-27 
* New: Added Toolbar integration.
* Update: Improved translation loading.
* Update: Minor code and readme adjustments :).
* Update: Updated German translations and also the .pot file for all translators.



### 🎉 1.0.0 - 2013-07-16 
* Initial release
* *NOTE:* This plugin is only for Genesis Version 1.9.x or higher!


## Upgrade Notice 

### 1.3.4
Now with WordPress 5.2 Site Health integration. Plus few internal improvements. - **Update highly recommended!**

### 1.3.3
Maintenance release. Bundled library update. Small tweaks. Update recommended.

### 1.3.2
Maintenance release. Bundled library update. Small tweaks. Update recommended.


### 1.3.1
New "At a Glance" meta box on Genesis Settings. Plus more tweaks and improvements.


### 1.3.0 
New "At a Glance" meta box on Genesis Settings. Plus more tweaks and improvements.


### 1.2.0 
Maintenance release. Make the plugin work again.


### 1.1.1 
Fixed PHP Notice. Also updated .pot file for translators plus German translations.


### 1.1.0 
Maintenance release: Added Toolbar integration. Some internal improvements and updates. Also updated .pot file for translators plus German translations.


### 1.0.0 
Just released into the wild.


## Donate 
Enjoy using *Genesis What's New Info*? Please consider [making a small donation](https://www.paypal.me/deckerweb) to support the project's continued development.


## Idea Behind / Philosophy 
Really simple stuff: bring a link to an already existing Genesis admin page to the left-hand admin menu. Result: "Genesis > What's New"


## Credits 
* Thanks to [Gary Jones @garyj](https://twitter.com/garyj) for initial idea to create this plugin! Follow him and support his work.


## Last but not least 
**Special Thanks go out to my family for allowing me to do such spare time projects (aka free plugins) and supporting me in every possible way!**


CHANGES.md
# Genesis What's New Info

**General Info**

* [Plugin page on WordPress.org: wordpress.org/plugins/genesis-whats-new-info/](https://wordpress.org/plugins/genesis-whats-new-info/)
* [Translate the plugin](https://translate.wordpress.org/projects/wp-plugins/genesis-whats-new-info)
* [**Join plugin's newsletter for insider info, tutorials and more**](https://eepurl.com/gbAUUn)
* [**Become a Patron**](https://www.patreon.com/deckerweb) and support ongoing development, maintenance and support of this plugin
* [**Donate** for the further development & support of the plugin](https://www.paypal.me/deckerweb)
* [Plugin's documentation & FAQ](https://wordpress.org/plugins/genesis-whats-new-info/#faq)
* [Facebook Community User Group](https://www.facebook.com/groups/deckerweb.wordpress.plugins/)
* [Facebook Info Page for Deckerweb Plugins](https://www.facebook.com/deckerweb.wordpress.plugins/)


## Changelog of the Plugin

### ⚡ 1.3.4 - 2019-05-07

* *New: Successfully tested with WordPress 5.2*
* New: Integrated with WordPress 5.2+ new Site Health feature: Genesis What's New Info now has an extra section on the Debug Info tab - this is especially helpful for support requests
* New: Added Genesis Developer Docs to Metabox
* Tweak: Updated Genesis Changelog URL
* Tweak: Updated bundled library DDWlib Plugin Installer Recommendations to latest version (v1.4.1) - feature updates
* Tweak: Updated `.pot` file plus all German translations (formal, informal) and language packs
* New: [Join my newsletter for DECKERWEB WordPress Plugins](https://eepurl.com/gbAUUn) - insider info, plus tutorials and more useful stuff


### ⚡ 1.3.3 - 2018-11-03

* New: [Video of plugin walkthrough and live demo](https://www.youtube.com/watch?v=w6CUGbHm5PI)
* Tweak: Updated bundled library DDWlib Plugin Installer Recommendations to latest version (v1.2.0) - which brings enhanced CSS styles, including for the "Dark Mode" plugin
* Tweak: Few internal code tweaks and improvements


### ⚡ 1.3.2 - 2018-10-01

* New: Added plugin update message also to Plugins page (overview table)
* New: Created special [Facebook Group for user community support](https://www.facebook.com/groups/deckerweb.wordpress.plugins/) for all plugins from me (David Decker - DECKERWEB), this one here included! ;-) - [please join at facebook!](https://www.facebook.com/groups/deckerweb.wordpress.plugins/)
* Tweak: Updated bundled library DDWlib Plugin Installer Recommendations to latest version (v1.1.0) - which brings smaller additions and enhancements, like CSS styles to the upload areas and plugin cards, plus plugin version number on plugin cards
* Tweak: Small internal code improvements
* Tweak: Updated `.pot` file plus all German translations (formal, informal) and language packs


### ⚡ 1.3.1 - 2018-08-24

* Fix: Bug with wrong constant


### 🎉 1.3.0 - 2018-08-23 

* New: Added extra "At a Glance" meta box on Genesis Theme Settings with proper version info, and links to: What's New page, Changelog website, StudioPress Blog, #GenesisWP Facebook Group, #GenesisWP Slack Channel, #GenesisWP Tweets -- Note: this is fully optional, meta box can be removed via "Screen Options", and also via our filter/ WP Core function
* New: Added `composer.json` file to the plugin's root folder - this is great for developers using Composer
* New: Added `README.md` file for plugin's GitHub.com repository to make it more readable there
* Tweak: Simplied rendering of What's New Page again, hooray! :)
* Tweak: Changed all of the plugin's filter hook tags to new format!
* Tweak: Added plugins recommendations library by deckerweb to improve the plugin installer tips (old filter function got removed)
* Tweak: Updated `.pot` file plus all German translations (formal, informal) and language packs
* Tweak: Enhanced, improved and corrected Readme.txt file here - also tweaked FAQ entry


### 🎉 1.2.0 - 2018-04-20

* New: Release on GitHub.com as well (for issues, development etc.), see here: [https://github.com/deckerweb/genesis-whats-new-info](https://github.com/deckerweb/genesis-whats-new-info)
* New: New plugin banner and icon on wordpress.org
* Fix: Make it work again, requires the newest Genesis version 2.6.0 or higher.
* Tweak: A few subtle tweaks here and there.
* Tweak: Updated translations and `.pot` file.


### ⚡ 1.1.1 - 2014-02-02

* Fix: Fixed PHP notice.


### 🎉 1.1.0 - 2014-01-27

* New: Added Toolbar integration.
* Update: Improved translation loading.
* Update: Minor code and readme adjustments :).
* Update: Updated German translations and also the .pot file for all translators.


### 🎉 1.0.0 - 2013-07-16

* Initial release
* *NOTE:* This plugin is only for Genesis Version 1.9.x or higher!


includes/gnewi-admin-extras.php
<?php

// includes/gnewi-admin-extras

/**
 * Prevent direct access to this file.
 *
 * @since 1.0.0
 */
if ( ! defined( 'ABSPATH' ) ) {
	exit( 'Sorry, you are not allowed to access this file directly.' );
}


/**
 * Add Admin link to plugin page.
 *
 * @since 1.0.0
 *
 * @param array $gnewi_links (Default) Array of plugin action links.
 * @return string Admin page link.
 */
function ddw_gnewi_admin_page_link( $gnewi_links ) {

	/** Admin page link */
	$gnewi_admin_link = sprintf(
		'<a href="%1$s" title="%2$s"><span class="dashicons-before dashicons-info"></span> %3$s</a>',
		esc_url( admin_url( 'admin.php?page=genesis-upgraded' ) ),
		sprintf(
			/* translators: %s - Label of Admin page ("Genesis Whats's New") */
			esc_html__( 'Go to the %s admin page', 'genesis-whats-new-info' ),
			esc_html__( 'Genesis What\'s New', 'genesis-whats-new-info' )
		),
		esc_attr__( 'What\'s New', 'genesis-whats-new-info' )
	);

	/** Set the order of the links */
	array_unshift( $gnewi_links, $gnewi_admin_link );

	/** Display plugin settings links */
	return apply_filters( 'gnewi/filter/plugins_page/admin_link', $gnewi_links );

}  // end function


add_filter( 'plugin_row_meta', 'ddw_gnewi_plugin_links', 10, 2 );
/**
 * Add various support links to plugin page.
 *
 * @since 1.0.0
 * @since 1.3.0 Improved link building.
 *
 * @uses ddw_gnewi_info_values()
 * @uses ddw_gnewi_get_info_link()
 *
 * @param array  $gnewi_links (Default) Array of plugin meta links
 * @param string $gnewi_file  URL of base plugin file
 *
 * @return array $gnewi_links Array of plugin link strings to build HTML markup.
 */
function ddw_gnewi_plugin_links( $gnewi_links, $gnewi_file ) {

	/** Capability check */
	if ( ! current_user_can( 'install_plugins' ) ) {
		return $gnewi_links;
	}

	/** List additional links only for this plugin */
	if ( $gnewi_file === GNEWI_PLUGIN_BASEDIR . 'genesis-whats-new-info.php' ) {

		?>
			<style type="text/css">
				tr[data-plugin="<?php echo $gnewi_file; ?>"] .plugin-version-author-uri a.dashicons-before:before {
					font-size: 17px;
					margin-right: 2px;
					opacity: .85;
					vertical-align: sub;
				}
			</style>
		<?php

		/* translators: Plugins page listing */
		$gnewi_links[] = ddw_gnewi_get_info_link( 'url_translate', esc_html_x( 'Translations', 'Plugins page listing', 'genesis-whats-new-info' ), 'dashicons-before dashicons-translation' );

		/* translators: Plugins page listing */
		$gnewi_links[] = ddw_gnewi_get_info_link( 'url_fb_group', esc_html_x( 'Facebook Group', 'Plugins page listing', 'genesis-whats-new-info' ), 'dashicons-before dashicons-facebook' );

		/* translators: Plugins page listing */
		$gnewi_links[] = ddw_gnewi_get_info_link( 'url_donate', esc_html_x( 'Donate', 'Plugins page listing', 'genesis-whats-new-info' ), 'button dashicons-before dashicons-thumbs-up' );

		/* translators: Plugins page listing */
		$gnewi_links[] = ddw_gnewi_get_info_link( 'url_newsletter', esc_html_x( 'Join our Newsletter', 'Plugins page listing', 'genesis-whats-new-info' ), 'button-primary dashicons-before dashicons-awards' );

	}  // end-if plugin links

	/** Output the links */
	return apply_filters(
		'gnewi/filter/plugins_page/more_links',
		$gnewi_links
	);

}  // end function


add_filter( 'debug_information', 'ddw_gnewi_site_health_add_debug_info', 9 );
/**
 * Add additional plugin related info to the Site Health Debug Info section.
 *   (Only relevant for WordPress 5.2 or higher)
 *
 * @link https://make.wordpress.org/core/2019/04/25/site-health-check-in-5-2/
 *
 * @since 1.3.4
 *
 * @param array $debug_info Array holding all Debug Info items.
 * @return array Modified array of Debug Info.
 */
function ddw_gnewi_site_health_add_debug_info( $debug_info ) {

	$string_undefined = esc_html__( 'Undefined', 'genesis-whats-new-info' );

	/** Add our Debug info */
	$debug_info[ 'genesis-whats-new-info' ] = array(
		'label'  => esc_html__( 'Genesis What\'s New Info', 'genesis-whats-new-info' ) . ' (' . esc_html__( 'Plugin', 'genesis-whats-new-info' ) . ')',
		'fields' => array(
			'gwnf_plugin_version' => array(
				'label' => __( 'Plugin version', 'genesis-whats-new-info' ),
				'value' => GNEWI_VERSION,
			),
			'PARENT_THEME_VERSION' => array(
				'label' => 'Genesis: PARENT_THEME_VERSION',
				'value' => ( ! defined( 'PARENT_THEME_VERSION' ) ? $string_undefined : PARENT_THEME_VERSION ),
			),
			'CHILD_THEME_VERSION' => array(
				'label' => 'Genesis Child: CHILD_THEME_VERSION',
				'value' => ( ! defined( 'CHILD_THEME_VERSION' ) ? $string_undefined : CHILD_THEME_VERSION ),
			),
		),
	);

	/** Return modified Debug Info array */
	return $debug_info;

}  // end function


if ( ! function_exists( 'ddw_wp_site_health_remove_percentage' ) ) :

	add_action( 'admin_head', 'ddw_wp_site_health_remove_percentage', 100 );
	/**
	 * Remove the "Percentage Progress" display in Site Health feature as this will
	 *   get users obsessed with fullfilling a 100% where there are non-problems!
	 *
	 * @link https://make.wordpress.org/core/2019/04/25/site-health-check-in-5-2/
	 *
	 * @since 1.3.4
	 */
	function ddw_wp_site_health_remove_percentage() {

		/** Bail early if not on WP 5.2+ */
		if ( version_compare( $GLOBALS[ 'wp_version' ], '5.2-beta', '<' ) ) {
			return;
		}

		?>
			<style type="text/css">
				.site-health-progress {
					display: none;
				}
			</style>
		<?php

	}  // end function

endif;


if ( ! function_exists( 'ddw_genesis_tweak_plugins_submenu' ) ) :

	add_action( 'admin_menu', 'ddw_genesis_tweak_plugins_submenu', 11 );
	/**
	 * Add Genesis submenu redirecting to "genesis" plugin search within the
	 *   WordPress.org Plugin Directory. For Genesis 2.10.0 or higher this
	 *   replaces the "Genesis Plugins" submenu which only lists plugins from
	 *   StudioPress - but there are many more from the community.
	 *
	 * @since 1.3.4
	 *
	 * @uses remove_submenu_page()
	 * @uses add_submenu_page()
	 */
	function ddw_genesis_tweak_plugins_submenu() {

		/** Remove the StudioPress plugins submenu */
		if ( class_exists( 'Genesis_Admin_Plugins' ) ) {
			remove_submenu_page( 'genesis', 'genesis-plugins' );
		}

		/** Add a Genesis community plugins submenu */
		add_submenu_page(
			'genesis',
			esc_html__( 'Genesis Plugins from the Plugin Directory', 'genesis-whats-new-info' ),
			esc_html__( 'Genesis Plugins', 'genesis-whats-new-info' ),
			'install_plugins',
			esc_url( network_admin_url( 'plugin-install.php?s=genesis&tab=search&type=term' ) )
		);

	}  // end function

endif;


/**
 * Inline CSS fix for Plugins page update messages.
 *
 * @since 1.3.2
 *
 * @see ddw_gnewi_plugin_update_message()
 * @see ddw_gnewi_multisite_subsite_plugin_update_message()
 */
function ddw_gnewi_plugin_update_message_style_tweak() {

	?>
		<style type="text/css">
			.gnewi-update-message p:before,
			.update-message.notice p:empty {
				display: none !important;
			}
		</style>
	<?php

}  // end function


add_action( 'in_plugin_update_message-' . GNEWI_PLUGIN_BASEDIR . 'genesis-whats-new-info.php', 'ddw_gnewi_plugin_update_message', 10, 2 );
/**
 * On Plugins page add visible upgrade/update notice in the overview table.
 *   Note: This action fires for regular single site installs, and for Multisite
 *         installs where the plugin is activated Network-wide.
 *
 * @since 1.3.2
 *
 * @param object $data
 * @param object $response
 * @return string Echoed string and markup for the plugin's upgrade/update
 *                notice.
 */
function ddw_gnewi_plugin_update_message( $data, $response ) {

	if ( isset( $data[ 'upgrade_notice' ] ) ) {

		ddw_gnewi_plugin_update_message_style_tweak();

		printf(
			'<div class="update-message gnewi-update-message">%s</div>',
			wpautop( $data[ 'upgrade_notice' ] )
		);

	}  // end if

}  // end function


add_action( 'after_plugin_row_wp-' . GNEWI_PLUGIN_BASEDIR . 'genesis-whats-new-info.php', 'ddw_gnewi_multisite_subsite_plugin_update_message', 10, 2 );
/**
 * On Plugins page add visible upgrade/update notice in the overview table.
 *   Note: This action fires for Multisite installs where the plugin is
 *         activated on a per site basis.
 *
 * @since 1.3.2
 *
 * @param string $file
 * @param object $plugin
 * @return string Echoed string and markup for the plugin's upgrade/update
 *                notice.
 */
function ddw_gnewi_multisite_subsite_plugin_update_message( $file, $plugin ) {

	if ( is_multisite() && version_compare( $plugin[ 'Version' ], $plugin[ 'new_version' ], '<' ) ) {

		$wp_list_table = _get_list_table( 'WP_Plugins_List_Table' );

		ddw_gnewi_plugin_update_message_style_tweak();

		printf(
			'<tr class="plugin-update-tr"><td colspan="%s" class="plugin-update update-message notice inline notice-warning notice-alt"><div class="update-message gnewi-update-message"><h4 style="margin: 0; font-size: 14px;">%s</h4>%s</div></td></tr>',
			$wp_list_table->get_column_count(),
			$plugin[ 'Name' ],
			wpautop( $plugin[ 'upgrade_notice' ] )
		);

	}  // end if

}  // end function


/**
 * Optionally tweaking Plugin API results to make more useful recommendations to
 *   the user.
 *
 * @since 1.2.0
 * @since 1.3.0 Complete refactoring, using library class DDWlib Plugin
 *              Installer Recommendations
 */

add_filter( 'ddwlib_plir/filter/plugins', 'ddw_gnewi_register_plugin_recommendations' );
/**
 * Register specific plugins for the class "DDWlib Plugin Installer
 *   Recommendations".
 *   Note: The top-level array keys are plugin slugs from the WordPress.org
 *         Plugin Directory.
 *
 * @since 1.3.0
 *
 * @param array $plugins Array holding all plugin recommendations, coming from
 *                       the class and the filter.
 * @return array Filtered and merged array of all plugin recommendations.
 */
function ddw_gnewi_register_plugin_recommendations( array $plugins ) {
  
  	/** Remove our own slug when we are already active :) */
  	if ( isset( $plugins[ 'genesis-whats-new-info' ] ) ) {
  		$plugins[ 'genesis-whats-new-info' ] = null;
  	}

  	/** Register our additional plugin recommendations */
	$gnewi_plugins = array(
		'genesis-layout-extras' => array(
			'featured'    => 'yes',
			'recommended' => 'yes',
			'popular'     => 'yes',
		),
		'genesis-widgetized-footer' => array(
			'featured'    => 'yes',
			'recommended' => 'yes',
			'popular'     => 'yes',
		),
		'genesis-widgetized-notfound' => array(
			'featured'    => 'yes',
			'recommended' => 'yes',
			'popular'     => 'yes',
		),
		'genesis-widgetized-archive' => array(
			'featured'    => 'yes',
			'recommended' => 'yes',
			'popular'     => 'no',
		),
		'blox-lite' => array(
			'featured'    => 'yes',
			'recommended' => 'yes',
			'popular'     => 'no',
		),
		'genesis-title-toggle' => array(
			'featured'    => 'yes',
			'recommended' => 'yes',
			'popular'     => 'yes',
		),
		'genesis-footer-builder' => array(
			'featured'    => 'yes',
			'recommended' => 'yes',
			'popular'     => 'no',
		),
		'display-featured-image-genesis' => array(
			'featured'    => 'yes',
			'recommended' => 'yes',
			'popular'     => 'no',
		),
		'genesis-enews-extended' => array(
			'featured'    => 'no',
			'recommended' => 'no',
			'popular'     => 'yes',
		),
		'genesis-simple-edits' => array(
			'featured'    => 'no',
			'recommended' => 'no',
			'popular'     => 'yes',
		),
		'genesis-simple-sidebars' => array(
			'featured'    => 'no',
			'recommended' => 'yes',
			'popular'     => 'yes',
		),
		'genesis-simple-hooks' => array(
			'featured'    => 'no',
			'recommended' => 'no',
			'popular'     => 'yes',
		),
	);

  	/** Merge with the existing recommendations and return */
	return array_merge( $plugins, $gnewi_plugins );

}  // end function

/** Optionally add string translations for the library */
if ( ! function_exists( 'ddwlib_plir_strings_plugin_installer' ) ) :

	add_filter( 'ddwlib_plir/filter/strings/plugin_installer', 'ddwlib_plir_strings_plugin_installer' );
	/**
	 * Optionally, make strings translateable for included library "DDWlib Plugin
	 *   Installer Recommendations".
	 *   Strings:
	 *    - "Newest" --> tab in plugin installer toolbar
	 *    - "Version:" --> label in plugin installer plugin card
	 *
	 * @since 1.3.2
	 * @since 1.3.4 Added new strings.
	 *
	 * @param array $strings Holds all filterable strings of the library.
	 * @return array Array of tweaked translateable strings.
	 */
	function ddwlib_plir_strings_plugin_installer( $strings ) {

		$strings[ 'newest' ] = _x(
			'Newest',
			'Plugin installer: Tab name in installer toolbar',
			'genesis-whats-new-info'
		);

		$strings[ 'version' ] = _x(
			'Version:',
			'Plugin card: plugin version',
			'genesis-whats-new-info'
		);

		$strings[ 'ddwplugins_tab' ] = _x(
			'deckerweb Plugins',
			'Plugin installer: Tab name in installer toolbar',
			'genesis-whats-new-info'
		);

		$strings[ 'tab_title' ] = _x(
			'deckerweb Plugins',
			'Plugin installer: Page title',
			'genesis-whats-new-info'
		);

		$strings[ 'tab_slogan' ] = __( 'Great helper tools for Site Builders to save time and get more productive', 'genesis-whats-new-info' );

		$strings[ 'tab_info' ] = sprintf(
			__( 'You can use any of our free plugins or premium plugins from %s', 'genesis-whats-new-info' ),
			'<a href="https://deckerweb-plugins.com/" target="_blank" rel="nofollow noopener noreferrer">' . $strings[ 'tab_title' ] . '</a>'
		);

		$strings[ 'tab_newsletter' ] = __( 'Join our Newsletter', 'genesis-whats-new-info' );

		$strings[ 'tab_fbgroup' ] = __( 'Facebook User Group', 'genesis-whats-new-info' );

		return $strings;

	}  // end function

endif;  // function check

/** Include class DDWlib Plugin Installer Recommendations */
require_once( GNEWI_PLUGIN_DIR . 'includes/ddwlib-plir/ddwlib-plugin-installer-recommendations.php' );
